exchanges/credentials.go

```diff
@@ -0,0 +1,428 @@
+	}
+
+	// Check to see if the user has enabled AuthenticatedSupport, but has
+	// invalid API credentials set and loaded by config
+	return b.ValidateAPICredentials(creds)
+}
+
+// AreCredentialsValid returns if the supplied credentials are valid.
+func (b *Base) AreCredentialsValid(ctx context.Context) bool {
+	creds, err := b.GetCredentials(ctx)
+	return err == nil && b.ValidateAPICredentials(creds) == nil
+}
+
+// GetDefaultCredentials returns the exchange.Base api credentials loaded by
+// config.json
+func (b *Base) GetDefaultCredentials() *Credentials {
+	b.API.credMu.RLock()
+	defer b.API.credMu.RUnlock()
+	if b.API.credentials == nil {
+		return nil
+	}
+	creds := *b.API.credentials
+	return &creds
+}
+
+// GetCredentials checks and validates current credentials, context credentials
+// override default credentials, if no credentials found, will return an error.
+func (b *Base) GetCredentials(ctx context.Context) (*Credentials, error) {
+	value := ctx.Value(contextCrendentialsFlag)
+	if value != nil {
+		ctxCredStore, ok := value.(*contextCredentialsStore)
+		if !ok {
+			return &Credentials{}, errContextCredentialsFailure
+		}
+
+		creds := ctxCredStore.Get()
+		if err := b.CheckCredentials(creds, true); err != nil {
+			return creds, fmt.Errorf("context credentials issue: %w", err)
+		}
+		return creds, nil
+	}
+
+	err := b.CheckCredentials(b.API.credentials, false)
+	if err != nil {
+		return &Credentials{}, err
+	}
+	subAccountOverride, ok := ctx.Value(contextSubAccountFlag).(string)
+	b.API.credMu.RLock()
+	defer b.API.credMu.RUnlock()
+	creds := *b.API.credentials
+	if ok {
+		creds.SubAccount = subAccountOverride
+	}
+	return &creds, nil
+}
+
+// ValidateAPICredentials validates the exchanges API credentials
+func (b *Base) ValidateAPICredentials(creds *Credentials) error {
+	b.API.credMu.RLock()
+	defer b.API.credMu.RUnlock()
+	if creds.IsEmpty() {
+		return fmt.Errorf("%s %w", b.Name, ErrCredentialsAreEmpty)
+	}
+	if b.API.CredentialsValidator.RequiresKey &&
+		(creds.Key == "" || creds.Key == config.DefaultAPIKey) {
+		return fmt.Errorf("%s %w", b.Name, errRequiresAPIKey)
+	}
+
+	if b.API.CredentialsValidator.RequiresSecret &&
+		(creds.Secret == "" || creds.Secret == config.DefaultAPISecret) {
+		return fmt.Errorf("%s %w", b.Name, errRequiresAPISecret)
+	}
+
+	if b.API.CredentialsValidator.RequiresPEM &&
+		(creds.PEMKey == "" || strings.Contains(creds.PEMKey, "JUSTADUMMY")) {
+		return fmt.Errorf("%s %w", b.Name, errRequiresAPIPEMKey)
+	}
+
+	if b.API.CredentialsValidator.RequiresClientID &&
+		(creds.ClientID == "" || creds.ClientID == config.DefaultAPIClientID) {
+		return fmt.Errorf("%s %w", b.Name, errRequiresAPIClientID)
+	}
+
+	if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !b.LoadedByConfig {
+		_, err := crypto.Base64Decode(creds.Secret)
+		if err != nil {
+			return fmt.Errorf("%s API secret %w: %s", b.Name, errBase64DecodeFailure, err)
+		}
+	}
```

Comment:
This is a bit more of a todo, but take the following example:
```
go run .  --apikey="LOL" --apisecret="SECRET" --apisubaccount="GLORIOUSCODE" --timeout=5m getaccountinfo binance spot
```
But we don't support subaccounts for Binance, this could given an impression that it is the subaccounts data, when it isn't.
A future peice of work could be to have a `SupportsSubAccount` flag, so the validation could then change to `subAccount != "" && !supportsSubAccount { errValidationStuff` }`